src/DominantJuice.sol

```diff
@@ -0,0 +1,396 @@
+import {IJBPayDelegate3_1_1} from "@jbx-protocol/juice-contracts-v3/contracts/interfaces/IJBPayDelegate3_1_1.sol";
+import {IJBPaymentTerminal} from "@jbx-protocol/juice-contracts-v3/contracts/interfaces/IJBPaymentTerminal.sol";
+import {IJBRedemptionDelegate3_1_1} from
+    "@jbx-protocol/juice-contracts-v3/contracts/interfaces/IJBRedemptionDelegate3_1_1.sol";
+import {IJBSingleTokenPaymentTerminal} from
+    "@jbx-protocol/juice-contracts-v3/contracts/interfaces/IJBSingleTokenPaymentTerminal.sol";
+import {IJBSingleTokenPaymentTerminalStore3_1_1} from
+    "@jbx-protocol/juice-contracts-v3/contracts/interfaces/IJBSingleTokenPaymentTerminalStore3_1_1.sol";
+import {JBDidPayData3_1_1} from "@jbx-protocol/juice-contracts-v3/contracts/structs/JBDidPayData3_1_1.sol";
+import {JBDidRedeemData3_1_1} from "@jbx-protocol/juice-contracts-v3/contracts/structs/JBDidRedeemData3_1_1.sol";
+import {JBFundingCycle} from "@jbx-protocol/juice-contracts-v3/contracts/structs/JBFundingCycle.sol";
+import {JBPayDelegateAllocation3_1_1} from
+    "@jbx-protocol/juice-contracts-v3/contracts/structs/JBPayDelegateAllocation3_1_1.sol";
+import {JBPayParamsData} from "@jbx-protocol/juice-contracts-v3/contracts/structs/JBPayParamsData.sol";
+//import {JBDidRedeemData} from "@jbx-protocol/juice-contracts-v3/contracts/structs/JBDidRedeemData.sol";
+import {JBRedeemParamsData} from "@jbx-protocol/juice-contracts-v3/contracts/structs/JBRedeemParamsData.sol";
+import {JBRedemptionDelegateAllocation3_1_1} from
+    "@jbx-protocol/juice-contracts-v3/contracts/structs/JBRedemptionDelegateAllocation3_1_1.sol";
+import {JBTokenAmount} from "@jbx-protocol/juice-contracts-v3/contracts/structs/JBTokenAmount.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {UD60x18, ud, add, pow, powu, div, mul, wrap, unwrap} from "@prb/math/UD60x18.sol";
+
+/**
+ * @title Dominant Juice
+ * @author Armand Daigle
+ * @notice This contract is a JB Data Source, JB Pay Delegate, JB Redemption Delegate, and a Refund Bonus Escrow
+ * implementation. Campaigns, referred to as cycles from here on, that use this implementation as a Data Source
+ * will incorporate Alex Tabarrok's dominant assurance mechanism, which incentivize early pledgers with a campaign
+ * creator-provided refund bonus that can be withdrawn from this contract after a failed campaign.
+ * @dev This contract uses the JB Data Source, Pay Delegate, and Pay Redemption interfaces and extends the
+ * functionality of a Juicebox project. There is no constructor since each campaign will deploy a clone of this
+ * contract. The initialize() function acts as the constructor. This contract is written to only be used for one cycle.
+ * If projects want dominant assurance on more cycles, refactoring is required.
+ */
+contract DominantJuice is IJBFundingCycleDataSource3_1_1, IJBPayDelegate3_1_1, IJBRedemptionDelegate3_1_1, Ownable {
```

Comment:
FYI: I don't like Ownable and neither do auditors. AccessControl is a better approach

Reply:
K, good to know. I can replace. I've never used it. Are there any tricky things to know off the bat, or is it relatively straight forward to use?

Reply:
Digging into this more, do you recommend using AccessControlDefaultAdminRules.sol? And what roles do you think we should have? 

- Default admin role - one or both of us? This role would not have any rights, other than being admin for other roles.
- Project Creator role for multisig, which is only role that can call depositBonus() and CreatorWithdraw().
- What else? Backup admin role?

Reply:
There's usually a super-admin role that can change other roles by default. I think there's a `CampaignManager` role that executes `depositBonus` and `creatorWithdraw`. I like Pausable contracts, so I tend to have a `Pauser` and if you're using upgradeable contracts, you'd have an `Upgrader`.